clmm/modbackend/clmm_modeling.py

```diff
@@ -0,0 +1,36 @@
+# CLMModeling abstract class
```

Comment:
If this parent class does not has any pre-set function, is it worth it to keep it? It just serve as an API template as it is.

Reply:
Using an abstract base class is good to control the subsequent implementations, take a look at [PEP 3119](https://www.python.org/dev/peps/pep-3119). Here we can put all documentation and common implementation of methods. We should actually make it better by using ABC in clmm_modeling.py .

Reply:
I think another way to phrase this is that we need it to support the back-ends just because they have modeling objects that behave differently, similarly to why we need a cosmology object (#293).